kernel/exec.c

```diff
@@ -994,6 +989,26 @@ int myst_exec(
     if (__myst_kernel_args.debug_symbols)
         ECHECK(_add_crt_symbols(crt_data, crt_size));
 
+    /* remove the thread clone flag CLONE_VFORK flag as this is used to
+     * determine if we do CRT cleanup. As we are re-initializing the CRT it is
+     * safe to re-enable the cleanup.
+     * At the same time we may need to wake the parent processes thread that
+     * launched us in the case we were created in the fork/exec wait mode. */
+    if (thread->clone.flags & CLONE_VFORK)
+    {
+        myst_thread_t* parent_launch_thread = myst_find_process_specific_thread(
+            thread->clone.vfork_parent_pid, thread->clone.vfork_parent_tid);
+        if (parent_launch_thread)
+        {
+            __sync_val_compare_and_swap(
+                &parent_launch_thread->fork_exec_futex_wait, 0, 1);
```

Comment:
Do we need to update the futex value to wake up the waiter?

Reply:
Why don't we put the futex variable in the child?

Reply:
We set the futex value to 1 and trigger

Reply:
I think setting the value to 1 is to avoid future futex waiters on the same address to get stuck. It doesn't make a difference in our simple scenario here.

Reply:
If we put the futex variable in the child, we don't have to search for the parent.

Reply:
setting it to one is to make sure it has not already been triggered before we hit the wait

Reply:
There is just 1 waiter and it only waits once. And when it waits, it sets the variable to 0 first.